pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/SanitizationTransformer.java

```diff
@@ -38,14 +41,22 @@
  * {@link FieldSpec}.
  */
 public class SanitizationTransformer implements RecordTransformer {
+  private static final String NULL_CHARACTER = "\0";
   private final Map<String, Integer> _stringColumnMaxLengthMap = new HashMap<>();
+  private final boolean _failOnTrimmedStringLength;
 
-  public SanitizationTransformer(Schema schema) {
+  public SanitizationTransformer(TableConfig tableConfig, Schema schema) {
     for (FieldSpec fieldSpec : schema.getAllFieldSpecs()) {
       if (!fieldSpec.isVirtualColumn() && fieldSpec.getDataType() == DataType.STRING) {
         _stringColumnMaxLengthMap.put(fieldSpec.getName(), fieldSpec.getMaxLength());
       }
     }
+    IngestionConfig ingestionConfig = tableConfig.getIngestionConfig();
+    if (ingestionConfig != null) {
+      _failOnTrimmedStringLength = ingestionConfig.isFailOnTrimmedStringLength();
```

Comment:
We already have a config `continueOnError`, should we use that one to control this as well? We probably don't want too fine grained control on failures

Reply:
Yes I thought of using the same but the present default value of `continueOnError` is false which would make this change backward-incompatible as instead of trimming and moving forward; we will have to fail the transform function. 
The behaviour of `continueOnError` is a bit opposite to what we need to maintain backward compatibility.

Reply:
I see. I misunderstood the intention of the new config. Basically you want to fail the ingestion when the value length is longer than the max length.

I can see we might want different mode when the value is over the max length, and it could be per column based:
- Trim (current behavior)
- Throw exception (added behavior in this PR)
- Fill default value (very useful for JSON/BYTES)

Based on this, I'd suggest adding a mode into `FieldSpec` which specifies the strategy of handling max length

Reply:
Hmm this sounds like a good suggestion for having columnar level flexibility! Let me update this PR based on the suggestion.

Reply:
Thanks @Jackie-Jiang for the review! Updated the patch accordingly.